app/blueprints/shared/helpers.py

```diff
@@ -136,3 +137,67 @@ def get_value_from_request(parameter_names) -> str | None:
         )
         if value:
             return value
+
+
+@dataclass
```

Comment:
Really nice work on this!  Could maybe move this to the models dir rather than on helpers.py?  Also maybe could unit test the methods